impeller/renderer/backend/vulkan/gpu_tracer_vk.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/vulkan/gpu_tracer_vk.h"
+
+#include <utility>
+#include "fml/trace_event.h"
+#include "impeller/base/validation.h"
+#include "impeller/renderer/backend/vulkan/command_buffer_vk.h"
+#include "impeller/renderer/backend/vulkan/command_encoder_vk.h"
+#include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "impeller/renderer/command_buffer.h"
+
+namespace impeller {
+
+static constexpr uint32_t kPoolSize = 128u;
+
+GPUTracerVK::GPUTracerVK(const std::shared_ptr<ContextVK>& context)
+    : context_(context) {
+  timestamp_period_ =
+      context_->GetPhysicalDevice().getProperties().limits.timestampPeriod;
```

Comment:
I think you need to check if this is actually supported right? Otherwise the value might be meaningless.

Reply:
the cmd itself is in 1.0, but I need to check that the queue supports timestamp queries. I think this will always be supported on graphics queues though, just might not have as much resolution as we'd like.

Reply:
some tests failed due to queue contruction issues so I added some logs to see what this value ends up as ...

Reply:
yup, this can be zero. Good call.

Reply:
Fixed